### PR TITLE
Product listing

### PR DESCRIPTION
**Pull request by teja538**
- added filter by categories functionality
- filter by price
- added clear filters functionality
- used useContext

preview link - https://rad-longma-bee0dc.netlify.app/

### CODE REVIEW

File: src/back-end/products-db.js [hunk 1]
```diff
@@ -124,9 +124,7 @@ const productsList =
         "rating":3.7,
         "price":7000,
         "category":"bangles",  
-    },
-   
-    
+    },   
 ]
```
Comment (sashaboi) [inline src/back-end/products-db.js:134]: could try adding a product for the 5000 slab of the price range filter so it doesnt show up an empty product section.

File: src/pages/product-listing/product-listing.jsx [hunk 2]
```diff
@@ -1,37 +1,71 @@
...
+    const priceFilteredProducts = getFilterByPrice(ratingFilteredProducts,state.price)
+    const finalProducts = getSortBy(priceFilteredProducts,state.sortBy)
+    console.log(finalProducts)
     return(
         <div className="prod-page">
             <div className="aside">
                 <div className="filter">
                     <h3>Filters</h3>
-                    <p>Clear</p>
+                    <button onClick={() => dispatch({type:"CLEAR-FILTER"})}>Clear</button>
```
Comment (sashaboi) [inline src/pages/product-listing/product-listing.jsx]: button could use a little styling , it still seems to be the default button

**Review by sashaboi — APPROVED** [review]
Looks amazing , great job! 

A suggestion not from this PR :  try adding on hover properties for buttons , and inputs in the filter section also the add to cart button. could help with better UX

**Comment by teja538** [comment]
> Looks amazing , great job!
> 
> A suggestion not from this PR : try adding on hover properties for buttons , and inputs in the filter section also the add to cart button. could help with better UX

sure onkar! thank you for suggestion will add that

File: src/back-end/products-db.js (hunk 1 above)
Comment (teja538) [reply src/back-end/products-db.js:134]: yeah! will do that

File: src/pages/product-listing/product-listing.jsx (hunk 2 above)
Comment (teja538) [reply src/pages/product-listing/product-listing.jsx]: done

**Review by dusariudaykumar — APPROVED** [review]
- the folder structure is also good.
- you used good variable names.
- overall it is good.

**Pull request merged**